### PR TITLE
Improvements to DockerLatentBuildSlave

### PR DESCRIPTION
**Pull request by tardyp**
-docker image is now renderable

This actually pull some dependencies, notabily the removal of build_status.properties, so that properties are available at the time of buildslave preparation.

- environment variables are passed to the slave to explain it how to connect to the master, and avoid hardcoding slavename and slave pass in the dockerfile

- some example dockerfiles

### CODE REVIEW

File: master/contrib/docker/slave/buildbot.tac [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+basedir = os.path.abspath(os.path.dirname(__file__))
+application = service.Application('buildslave')
+
+from twisted.python.log import ILogObserver, FileLogObserver
+
+application.setComponent(ILogObserver, FileLogObserver(sys.stdout).emit)
+# and slave on the same process!
+print os.environ
+buildmaster_host = os.environ.get("BUILDMASTER", 'localhost')
+port = int(os.environ.get("BUILDMASTER_PORT", 19989))
```
Comment (sa2ajj) [inline master/contrib/docker/slave/buildbot.tac]: For this to work `BUILDMASTER_PORT` must not be present, not just merely empty.

File: master/buildbot/buildslave/docker.py [hunk 2]
```diff
@@ -101,14 +106,26 @@ def __init__(self, name, password, docker_host, image=None, command=None,
         if tls is not None:
             self.client_args['tls'] = tls
 
+    def createEnvironment(self):
+        port = ""
+        if self.registration is not None:
+            port = str(self.registration.getPBPort())
+        return {
+            "BUILDMASTER": self.masterFQDN,
+            "BUILDMASTER_PORT": port,
```
Comment (sa2ajj) [inline master/buildbot/buildslave/docker.py]:
```
result = {
 ...
}
if self.registration is not None:
   result['BUILDMASTER_PORT'] = str(self.registration.getPBPort())
return result
```

**Comment by sa2ajj** [comment]
LGTM, otherwise (I'd a couple of prying eyes more)

File: master/docs/manual/cfg-buildslaves-docker.rst [hunk 3]
```diff
@@ -113,6 +113,33 @@ When your Dockerfile is ready, you can build your first image using the followin
...
+
+``pythonnode_slave``
+    a slave with python and node installed, which demonstrate how to reuse the base slave to create variations of build environments.
+
+The master setups several environment variables before starting the buildslaves:
+
+```BUILDMASTER``
+    The address of the master the slave shall connect to
+
+```BUILDMASTER_PORT``
```
Comment (djmitche) [inline master/docs/manual/cfg-buildslaves-docker.rst]: triple-quoted (as with the previous paragraph)

File: master/docs/relnotes/index.rst [hunk 4]
```diff
@@ -42,6 +46,11 @@ Features
 * The Buildbot slave now includes the number of CPUs in the information it supplies to the master on connection.
   This value is autodetected, but can be overridden with the ``--numcpus`` argument to ``buildslave create-slave``.
 
+* The :class:`DockerLatentBuildSlave` image attribute is now renderable (can take properties in account).
```
Comment (djmitche) [inline master/docs/relnotes/index.rst]: This is cool, and should be mentioned somewhere in the configuration section of the manual

File: master/docs/relnotes/index.rst [hunk 5]
```diff
@@ -42,6 +46,11 @@ Features
...
+
+* The :class:`DockerLatentBuildSlave` sets environment variables describing how to connect to the master.
+  Example dockerfiles can be found in ``master/contrib/docker``. (buildbot 0.9.0b2 is hardcoded in the docker file, needs to be updated when publishing 0.9.0)
```
Comment (djmitche) [inline master/docs/relnotes/index.rst]: Better to fix this hard-coding now, I think

**Comment by djmitche** [comment]
LGTM with those minor fixes

**Comment by tardyp** [comment]
oups.. I forgot about this one...

**Pull request merged**